src/Aspire.Hosting/VersionChecking/VersionCheckService.cs

```diff
@@ -0,0 +1,178 @@
+
+    private readonly IInteractionService _interactionService;
+    private readonly ILogger<VersionCheckService> _logger;
+    private readonly IConfiguration _configuration;
+    private readonly DistributedApplicationOptions _options;
+    private readonly IVersionFetcher _versionFetcher;
+    private readonly DistributedApplicationExecutionContext _executionContext;
+    private readonly TimeProvider _timeProvider;
+    private readonly SemVersion? _appHostVersion;
+
+    public VersionCheckService(IInteractionService interactionService, ILogger<VersionCheckService> logger,
+        IConfiguration configuration, DistributedApplicationOptions options, IVersionFetcher versionFetcher,
+        DistributedApplicationExecutionContext executionContext, TimeProvider timeProvider)
+    {
+        _interactionService = interactionService;
+        _logger = logger;
+        _configuration = configuration;
+        _options = options;
+        _versionFetcher = versionFetcher;
+        _executionContext = executionContext;
+        _timeProvider = timeProvider;
+
+        var version = typeof(VersionCheckService).Assembly.GetName().Version!;
+        var patch = version.Build > 0 ? version.Build : 0;
+        _appHostVersion = new SemVersion(version.Major, version.Minor, patch);
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        if (!_interactionService.IsAvailable || _executionContext.IsPublishMode || _configuration.GetBool(KnownConfigNames.VersionCheckDisabled, defaultValue: false))
+        {
+            // Don't check version if there is no way to prompt that information to the user.
+            // Or app is being run during a publish.
+            return;
+        }
+
+        try
+        {
+            await CheckForLatestAsync(stoppingToken).ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            // Ignore errors during shutdown.
+            if (!stoppingToken.IsCancellationRequested)
+            {
+                _logger.LogDebug(ex, "Error checking for latest version.");
+            }
+        }
+    }
+
+    private async Task CheckForLatestAsync(CancellationToken stoppingToken)
+    {
+        var now = _timeProvider.GetUtcNow();
+        var checkForLatestVersion = true;
+        if (_configuration[CheckDateKey] is string checkDateString &&
+            DateTime.TryParseExact(checkDateString, "o", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind, out var checkDate))
+        {
+            if (now - checkDate < s_checkInterval)
+            {
+                // Already checked within the last day.
+                checkForLatestVersion = false;
+            }
+        }
+
+        SemVersion? latestVersion = null;
+        if (checkForLatestVersion)
+        {
+            SecretsStore.TrySetUserSecret(_options.Assembly, CheckDateKey, now.ToString("o", CultureInfo.InvariantCulture));
```

Comment:
Do we want to use user-secrets for this? It feels like clutter to the user. What about a file in `IAspireStore` instead? The developer typically doesn't see the values in there.

Reply:
I don't know enough about app host/intergration development to say the pros vs cons.

@davidfowl @mitchdenny @eerhardt What do you think should be used here?

Reply:
Yeah user secrets is for the user to store user-specific config for the app, not the app to store config for itself. We should use `IAspireStore`.

Reply:
I looked at `IAspireStore` and it's for managing files. Not getting and setting name/values.

Reply:
I could imagine the AppHost managing a single "metadata" file where we could put more of this kind of information in the future. Similar to the `.aspire/settings.json` file for the CLI, but for the AppHost itself.

The `IAspireStore` would only be responsible for giving the `BasePath` directory of the store. Something else (probably internal) would know the file name, and how to (de)serialize information in and out of the metadata/settings file.

Reply:
All of this work to avoid a version in user secrets...

Reply:
> Note that the IAspireStore directory is meant to be cleaned up when someone `git clean`s. So maybe it shouldn't be used for the storage of "ignore version upgrade" bool? Unless we want to prompt them everytime they clean.

We don't want to reset the ignore version or last version check date on git clean.